### PR TITLE
Fix the problem that PR #99 accidentally removed code from PR #84, and GameState not synchronizing correctly

### PR DESCRIPTION
**Pull request by s3131212**
1. PR #99 accidentally remove the changes to asset server from PR #84 , this PR adds the changes back.
2. In multiprocess mode, GameState in the gateway services should be updated when received updates from Redis.

### CODE REVIEW

File: run/start-all.mjs [hunk 1]
```diff
@@ -60,11 +60,12 @@ async function main() {
     gatewayServers[serverName] = fork('../services/main/gateway-server.mjs', ['--service-name', serverName], { cwd: '.' });
   }
 
+  await new Promise(r => setTimeout(r, 4000)); // TODO: better way to prevent race condition.
```
Comment (s3131212) [inline run/start-all.mjs:63]: By the way, I think I need some help on this part. How do I ensure that standalone extension are executed **after** gateway services are fully started?
In some cases, a standalone extension might publish to Redis channel before gateway services are subscribe to that channel, causing the GameState to be out of sync.

Comment (ktpss95112) [reply run/start-all.mjs:63]: I think that in JS you should use an `async` function `launchGatewayService(someService)` to handle the startup of a service. In such way, you can do this:
```javascript
const allServices = [...];
const promises = allServices.forEach((s) => {return launchGatewayService(s)});
await Promise.all(promises);
```

I also saw you use `fork` to launch the gateway server. You may need a ping-pong message to know if your gateway server is ready.

File: run/start-all.mjs [hunk 2]
```diff
@@ -60,11 +60,25 @@ async function main() {
...
+      gatewayServers[serverName].on('message', msg => {
+        if(msg !== 'started') return;
+        console.log(`${serverName} has started successfully.`)
+        resolve(msg)
+      });
+      const timer = setTimeout(() => {
+        reject(new Error(`Promise timed out after 10 sec.`));
+      }, 10 * 1000);
+    });
+  });
```
Comment (ktpss95112) [inline run/start-all.mjs:75]:
```javascript
const primises = Object.entries(gatewayServers).map((entry) => {
    const [name, server] = entry;
    server.on(...);
    ...
})

```

Comment (ktpss95112) [reply run/start-all.mjs:75]: Error message should be more informative. For example, `Failed to start "${name}" server successfully in ${timeout} seconds.`.

**Review by ktpss95112 — APPROVED** [review]

**Pull request merged**